File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Consegna Legrottaglie Gionata/Kalk c++/calcolatrice/punto.cpp
#include "punto.h"
Punto::Punto():x(0),y(0){}
Punto::Punto(double a,double b):x(a),y(b){}
Punto::Punto(const Punto& p):x(p.x),y(p.y){}

double Punto:: get_X()const{return x;}
double Punto:: get_Y()const{return y;}

//overraiding operatori
bool Punto::operator==(const Punto& p)const{	return x == p.x && y == p.y;}
bool Punto::operator!=(const Punto& p)const{	return x != p.x || y != p.y;}
bool Punto::operator<(const Punto& p)const{
  if (y != p.y)
      return y < p.y;
  return x < p.x;
}

QString Punto::toString()const{
  return QString("(" + to_string_nozero(x) + "," + to_string_nozero(y)+ ")");
}
double distanza(const Punto& p1, const Punto& p2)
{
    const double dX = ((p1.get_X()-p2.get_X())*(p1.get_X()-p2.get_X()));
    const double dY = ((p1.get_Y()-p2.get_Y())*(p1.get_Y()-p2.get_Y()));
    return sqrt( dX + dY );
}

std::ostream& operator<<(std::ostream& os,const Punto& p){
        return os<<p.toString().toUtf8().constData();
}

QString to_string_nozero(double a){
  std::string s= std::to_string(a);
  int i=s.size()-1;
  bool out=false;
  while(s[i] != '.' && !out){
      if(s[i] == '0'){
        s.erase(s.size()-1,1);
        i--;
        }
      else
        out=true;
    }
  if(s[i]=='.') s.erase(s.size()-1,1);
  QString str = QString::fromStdString(s);
  return str;
}

File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Consegna Legrottaglie Gionata/Kalk Java/Punto.java

public class Punto {
	private double x, y;
	
	public Punto(double a, double b){		
		this.x=a;;
		this.y=b;
	}
	
	protected Punto(Punto p){
		this.x=p.x;
		this.y=p.y;
	}
	
	public double get_X() {
		return x;
	}
	public double get_Y() {
		return y;
	}

	public static double distanza(final Punto p1, final Punto p2){
		double dX = ((p1.get_X()-p2.get_X())*(p1.get_X()-p2.get_X()));
	    double dY = ((p1.get_Y()-p2.get_Y())*(p1.get_Y()-p2.get_Y()));
	    return Math.sqrt( dX + dY );
	}
	
	public boolean minore(Punto p) {
		if(this.y != p.y)
			return y < p.y;
		return x < p.x;
	}
	
	@Override public String toString() {
		return "(" + this.x + "," + this.y + ")";
	}
	
	
	@Override public boolean equals(Object obj) {
		if(!(obj instanceof Punto)) 
			return false;
		
		if(this == obj) 
			return true;
		Punto p = (Punto)obj;
		return p.x==x && p.y==y;
	}
	@Override
	public Punto clone(){
		return new Punto(this);
	}

}
